From 06a46ba755195810f2aeda01b12d1ccfe7c2dcfd Mon Sep 17 00:00:00 2001
From: Daiki Ueno <ueno@gnu.org>
Date: Mon, 28 Dec 2015 06:27:42 +0900
Subject: [PATCH] maint: fix operator precedence in mbrtowc test

This is a fix for test breakage introduced by commit 45228d96; the
equality expression must be parenthesized when negated with '!',
otherwise we always get:

  test-mbrtowc.c:49: assertion 'ret == (size_t)(-2)' failed

* m4/mbrtowc.m4 (gl_MBRTOWC_EMPTY_INPUT): Negate the entire expression.

Upstream-commit: 1f63650823cebf52044df840c81062ccb52163a2
Signed-off-by: Kamil Dudka <kdudka@redhat.com>
---
 gl/m4/mbrtowc.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gl/m4/mbrtowc.m4 b/gl/m4/mbrtowc.m4
index deb9f06..be2e9d6 100644
--- a/gl/m4/mbrtowc.m4
+++ b/gl/m4/mbrtowc.m4
@@ -569,7 +569,7 @@ changequote([,])dnl
            int
            main (void)
            {
-             return ! mbrtowc (&wc, "", 0, &mbs) == (size_t) -2;
+             return mbrtowc (&wc, "", 0, &mbs) != (size_t) -2;
            }]])],
         [gl_cv_func_mbrtowc_empty_input=yes],
         [gl_cv_func_mbrtowc_empty_input=no],
-- 
2.5.0

